### ISSUE TITLE
Add support for other Xbox Controllers

### ISSUE THREAD

**Issue by DrkCloudStrife** [delabrcd/rockband-drums-usb#4]
Hello, I was wondering if it would be possible to add multiple controller support for the USB Host Shield? Below you can find the Vendor ID as well as Product IDs of the controllers I would like to add support.

| Model | VendorID | ProductID |
|--------|--------|--------|
| 1914 | VID_045E | PID_0B12 | 
| 1797 | VID_045E | PID_0B00 | 
| 1698 | VID_045E | PID_02E3 | 
| 1708 | VID_045E | PID_02EA | 

I believe the only controller missing is the original Xbox One controller, which I do not have a hold of. But according to the internet, it's the model # 1537. Which should be the following:

<html>
<body>
<div style="box-sizing: border-box; color: rgb(51, 51, 51); font-family: &quot;Open Sans&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h5 class="+colour-primary" style="box-sizing: border-box; font-family: Nunito, sans-serif !important; font-weight: 500; line-height: 1.1; color: rgb(102, 199, 255); margin: 30px 0px 15px; font-size: 14px;">Device Details</h5><div class="details --type-device --auto-link" style="box-sizing: border-box;"><h3 class="details__heading" style="box-sizing: border-box; font-family: Nunito, sans-serif !important; font-weight: 500; line-height: 1.1; color: inherit; margin: 30px 0px 15px; font-size: 24px;">Xbox One Controller</h3><div style="box-sizing: border-box;"></div>

Type | Information
-- | --
ID | 02D1

</div></div>
</body>
</html>

<html>
<body>
<h5 class="+colour-primary" style="box-sizing: border-box; font-family: Nunito, sans-serif !important; font-weight: 500; line-height: 1.1; color: rgb(102, 199, 255); margin: 30px 0px 15px; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Vendor Details</h5><div class="details --type-vendor --auto-link" style="box-sizing: border-box; color: rgb(51, 51, 51); font-family: &quot;Open Sans&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 class="details__heading" style="box-sizing: border-box; font-family: Nunito, sans-serif !important; font-weight: 500; line-height: 1.1; color: inherit; margin: 30px 0px 15px; font-size: 24px;">Microsoft Corp.</h3><div style="box-sizing: border-box;"></div>

Type | Information
-- | --
ID | 045E

</div>
</body>
</html>

**Comment by delabrcd** [comment]
There should already be support for some of these: 
```
#define XBOX_ONE_PID1 0x02D1   // Microsoft X-Box One pad
#define XBOX_ONE_PID2 0x02DD   // Microsoft X-Box One pad (Firmware 2015)
#define XBOX_ONE_PID3 0x02E3   // Microsoft X-Box One Elite pad
#define XBOX_ONE_PID4 0x02EA   // Microsoft X-Box One S pad
#define XBOX_ONE_PID13 0x0B0A  // Microsoft X-Box One Adaptive Controller
#define XBOX_ONE_PID14 0x0B12  // Microsoft X-Box Core Controller
```

The only one I see missing is the 0B00 model. 

Can you re-confirm which ones don't currently work?

**Comment by DrkCloudStrife** [comment]
Oh, interesting. None of my controllers seem to work. Maybe my issue is with the USB HostShield. I know you mention to get the one RoHS, but even though the vendor claimed it was Android compatible, it doesn't have the labels you specified. I will try to get a different USB HostShield.

**Comment by DrkCloudStrife** [comment]
Closing this issue. I now know where to add missing controller PIDs. So I can open a PR, if needed, once I get the new USB Host Shield. Thanks!

**Comment by delabrcd** [comment]
Unfortunately those shields are finicky. 

I gave that general guideline going off of what I could see in pictures. 

The bad ones use 74HCT125 chips for both level translators, the good ones will follow the original design and use a 74HCT125 for the 5V side and a 74AHCT125 on the 3.3v side. 

I can say that I've ordered two from that AliExpress vendor linked in the readme (maybe a year apart) and had no issues. 

I'm considering just laying out my own board and selling that at cost just so that there isn't a need to deal with this. 

Edit: here's the schematic since it can be harder to find sometimes 
![Arduino-USB-2 0-Host-shield-schema](https://github.com/delabrcd/openrb-instruments/assets/36674601/77041951-f147-4dca-b5fd-edb0584da3f1)
